### PR TITLE
Unify uimenu behavior, part 1

### LINKED ISSUE #25131: Add "ESC" as closing option for "f" menu - "Draw what?"
**Is your feature request related to a problem? Please describe.**  
When you have empty hands "fists" and you press "f" a menu pops up which asks what to draw from a holster/sheath.
You can´t close this small menu with "ESC" which is inconvenient, as most other menues can be closed with "ESC".
Not having ESC for closing the menu means an interuption of gameplay flow. You always have to stop with what you do to figure out on the screen how to close the menu. Right now "1" or "ENTER" are a solution. But "ESC" has built up as like a muscle memory from the other menues, so it would be a good decision to implement that here.

**Describe the solution you'd like**  
As this feature, "ESC", was lately introduced to escape from the toolbelt menu (#25078), please do it for this menu also. 

**Reproduction**
1. Wield anything,
2. Have at least one bow on a sling or a gun in a holster.
3. Press "f"
4, Try to use "ESC".

**Additional context**  
draw what

### PR DESCRIPTION
**Pull request by Qrox**
#### Summary
SUMMARY: Infrastructure "Unify uimenu behavior"

#### Purpose of change
The `uimenu` interface is used throughout the game, but currently they lack consistent behavior. Some menus can be canceled while some menus cannot, or can only be canceled by selecting one of the entries, which may be at the beginning or at the end of the list. And many menus can be canceled with any unbound keys (bad for fat fingers). The legacy `menu`, `menu_vec`, and their compatibility constructors in `uimenu` have different behavior from the other `uimenu` interfaces (the default entry value starts from 1 instead of 0). And `uimenu` uses a single option `return_invalid` to accomplish three separate functionality: returning on selecting disabled entries; returning on unbound keys; allowing cancelling.

#### Describe the solution
This PR aims to improve and unify the behavior of the `uimenu` interface. It splits the option `return_invalid` into three seperate options: `allow_disabled`, `allow_anykey`, `allow_cancel`. The default has been changed to allow cancelling, disallow other unbound keys from cancelling, and disallow disabled entries from being selected, unless otherwise neccesary. It also unifies the default entry value to start from 0.

Some unused or rarely used legacy constructors and functions are removed. Other legacy constructors and functions are still supported for smooth migration.

Some menus are migrated in this PR for testing:
- Action menu
- ESC menu
- Bionic remote menu (only the parent menu)
- Draw weapon menu
- Sleep menu
- Reprogram manhack menu

#### Additional context
Closes #25131.

The reprogram manhack menu seems to be dead code currently: manhack can be reverted to item, so the code returns after asking the player whether to deactivate them, and the reprogram menu is never reached.

### CODE REVIEW

**Comment by Robik81** [comment]
I am currently fighting with uimenu strange behavior. Not sure if it is just Visual Studio compiler / Windows related or general problem.

If there is nested menu opened from 'parent' menu, it has to be set to 
`return_invalid = false;`
otherwise it closes immediately after it opens.

**Comment by Robik81** [comment]
Sure, I'l post the link once I make a PR for planting.

**Comment by kevingranade** [comment]
Please consider making uimenu handle all of return_invalid, allow_disabled, allow_anykey, allow_cancel, and migrating one menu at a time.  This will make testing, conflict resolution and review much easier.

**Comment by Robik81** [comment]
@Qrox I added the comment to relevant code line of
`return_invalid = false;` in my #25181 thread.

**Comment by mlangsdorf** [comment]
I'm with Kevin: let's get all the uimenus to work the same and do it in this PR.

I looked over the changes to the vehicle code and they looked sensible to me. All the menus and submenus were covered.

**Comment by Qrox** [comment]
I don't see an easy way to both support `return_invalid` and the new flags... The interface is changed, so it's hard to ensure that the old behavior of `return_invalid` will not change. For example, all the constructors now default to allowing cancelling, so old code that does not change the `return_invalid` flag will now get `UIMENU_CANCEL` as a return value, which might break things.

**Comment by kevingranade** [comment]
Don't change all the constructors then:
1. Make a new batch of constructors with the new behaviour (these can be wrappers around old constructors).
2. Port menus over to the new constructors one at a time. (over multiple PRs)
3. Remove the old constructors.

This ends up generating some additional boilerplate code in the short term, but makes the hard part, which is review and testing of the new code much easier.

File: src/ui.h [hunk 1]
```diff
@@ -193,16 +207,25 @@ class uimenu: public ui_container
...
         uimenu( bool cancelable, const char *mes,
                 const std::vector<std::string> options ); // legacy menu_vec
-        uimenu( bool cancelable, const char *mes, const std::vector<std::string> &options,
-                const std::string &hotkeys );
         uimenu( bool cancelable, int startx, int width, int starty, std::string title,
                 std::vector<uimenu_entry> ents );
-        uimenu( int startx, int width, int starty, std::string title, std::vector<uimenu_entry> ents );
+
+        // migration in process
+        uimenu( uimenu_migration_tag );
```
Comment (Qrox) [inline src/ui.h]: I have to add this migration tag so I can migrate the default constructions once at a time. In the final PR I'll remove this tag. I can't think of any better way to do this.

File: src/player.cpp [hunk 2]
```diff
@@ -7894,7 +7894,9 @@ bool player::dispose_item( item_location &&obj, const std::string& prompt )
 void player::mend_item( item_location&& obj, bool interactive )
 {
     if( g->u.has_trait( trait_DEBUG_HS ) ) {
-        uimenu menu( true, _( "Toggle which fault?" ) );
```
Comment (Qrox) [inline src/player.cpp]: This menu is not migrated, I just changed it to use the default constructor so I can remove one of the old constructors `uimenu::uimenu( bool, const char *, ... )`, which was not working properly anyway (it assigned neither the bool flag nor the message string to the corresponding class members)

**Pull request closed without merging**